service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/RequestStatementMapper.java

```diff
@@ -0,0 +1,202 @@
+    private static final String SPECIALTY = "Specialty: ";
+    private static final String REASON_CODE = "Reason Codes: ";
+    private static final String DEFAULT_REASON_CODE_XML = "<code code=\"3457005\" displayName=\"Patient referral\" codeSystem=\"2.16.840.1"
+        + ".113883.2.1.3.2.4.15\"/>";
+    private static final String NOTE = "Annotation: %s @ %s %s";
+    private static final String COMMA = ",";
+
+    private final MessageContext messageContext;
+
+    public String mapReferralRequestToRequestStatement(ReferralRequest referralRequest, boolean isNested) {
+        var requestStatementTemplateParameters = RequestStatementTemplateParameters.builder()
+            .requestStatementId(messageContext.getIdMapper().getOrNew(ResourceType.ReferralRequest, referralRequest.getId()))
+            .isNested(isNested)
+            .availabilityTime(StatementTimeMappingUtils.prepareAvailabilityTimeForReferralRequest(referralRequest))
+            .description(buildDescription(referralRequest))
+            .build();
+
+        if (!referralRequest.hasReasonCode()) {
+            requestStatementTemplateParameters.setDefaultReasonCode(DEFAULT_REASON_CODE_XML);
+        }
+
+        return TemplateUtils.fillTemplate(REQUEST_STATEMENT_TEMPLATE, requestStatementTemplateParameters);
+    }
+
+    private String buildDescription(ReferralRequest referralRequest) {
+        List<String> descriptionList = retrieveDescription(referralRequest);
+        return descriptionList.stream()
+            .filter(StringUtils::isNotEmpty)
+            .collect(Collectors.joining(StringUtils.SPACE));
+    }
+
+    private List<String> retrieveDescription(ReferralRequest referralRequest) {
+        return List.of(
+            buildIdentifierDescription(referralRequest),
+            buildPriorityDescription(referralRequest),
+            buildServiceRequestedDescription(referralRequest),
+            buildRequesterDescription(referralRequest),
+            buildSpecialtyDescription(referralRequest),
+            buildRecipientDescription(referralRequest),
+            buildReasonCodeDescription(referralRequest),
+            buildNoteDescription(referralRequest),
+            buildTextDescription(referralRequest)
+        );
+    }
+
+    private String buildIdentifierDescription(ReferralRequest referralRequest) {
+        Optional<String> identifier = Optional.empty();
+        if (referralRequest.hasIdentifier()) {
+            identifier = referralRequest.getIdentifier()
+                .stream()
+                .filter(val -> val.getSystem().equals(UBRN_SYSTEM_URL))
+                .map(Identifier::getValue)
+                .findFirst();
+        }
+        return identifier.map(value -> UBRN + value).orElse(StringUtils.EMPTY);
+    }
+
+    private String buildPriorityDescription(ReferralRequest referralRequest) {
+        if (referralRequest.hasPriority()) {
+            return PRIORITY + referralRequest.getPriority().getDisplay();
```

Comment:
This will most likely need to use the CodeableConceptMappingUtils::extractTextOrCoding "display or coding[0].display" but worth confirming

Reply:
From the HAPI FHIR ReferralRequest class, ReferralPriority is an Enum and not a Codeable concept.
Therefore we cannot use the CodeableConceptMappingUtils.